volume/drivers/extpoint.go

```diff
@@ -92,7 +92,9 @@ func Unregister(name string) bool {
 // driver with the given name has not been registered it checks if
 // there is a VolumeDriver plugin available with the given name.
 func lookup(name string) (volume.Driver, error) {
-	drivers.driverLock.Lock(name)
+	if err := drivers.driverLock.Lock(name); err != nil {
+		return nil, fmt.Errorf("Error looking up volume plugin %s: %v", name, err)
+	}
 	defer drivers.driverLock.Unlock(name)
```

Comment:
But the `defer` is after the `if err`.
If there is an error, it returns immediately at "line 96". No chance to run "line 98" `defer Unlock`.